python/python/lancedb/remote/table.py

```diff
@@ -307,10 +324,22 @@ def search(
             - and also the "_distance" column which is the distance between the query
             vector and the returned vector.
         """
-        if vector_column_name is None:
-            vector_column_name = inf_vector_column_query(self.schema)
-        query = LanceQueryBuilder._query_to_vector(self, query, vector_column_name)
-        return LanceVectorQueryBuilder(self, query, vector_column_name)
+        if vector_column_name is None and query is not None:
+            try:
```

Comment:
this feels wrong. Why can we pass `query_type = fts` in `vector search`?

Reply:
it's all-in-one search API, can do FTS and vector search, depending on the `query_type` and the type of `query`.
this is:
- if the vector_column isn't specified, then try to infer it
- if can't find such a vector column
    - if it's FTS, then it's fine
    - if it's vector search, we need to report error

Reply:
Can we change the top-level if statement to `if vector_column_name is None and query is not None and query_type != "fts":`?

Reply:
Yes that's good, Fixed